cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -4714,35 +4814,145 @@ macro(build_awssdk)
+  add_dependencies(AWS::aws-c-auth aws_c_auth_ep)
+
+  externalproject_add(aws_c_s3_ep
+                      ${EP_COMMON_OPTIONS}
+                      URL ${AWS_C_S3_SOURCE_URL}
+                      URL_HASH "SHA256=${ARROW_AWS_C_S3_BUILD_SHA256_CHECKSUM}"
+                      CMAKE_ARGS ${AWSSDK_COMMON_CMAKE_ARGS}
+                      BUILD_BYPRODUCTS ${AWS_C_S3_STATIC_LIBRARY}
+                      DEPENDS aws_c_common_ep
+                              aws_c_cal_ep
+                              s2n_ep
+                              aws_c_io_ep
+                              aws_c_http_ep
+                              aws_c_auth_ep)
+  add_dependencies(AWS::aws-c-s3 aws_c_s3_ep)
+
+  externalproject_add(aws_lc_ep
+                      ${EP_COMMON_OPTIONS}
+                      URL ${AWS_LC_SOURCE_URL}
+                      URL_HASH "SHA256=${ARROW_AWS_LC_BUILD_SHA256_CHECKSUM}"
+                      CMAKE_ARGS ${AWSSDK_COMMON_CMAKE_ARGS}
+                      BUILD_BYPRODUCTS ${CRYPTO_STATIC_LIBRARY} ${SSL_STATIC_LIBRARY})
+  add_dependencies(AWS::crypto aws_lc_ep)
+  add_dependencies(AWS::ssl aws_lc_ep)
+
+  externalproject_add(aws_crt_cpp_ep
+                      ${EP_COMMON_OPTIONS}
+                      URL ${AWS_CRT_CPP_SOURCE_URL}
+                      URL_HASH "SHA256=${ARROW_AWS_CRT_CPP_BUILD_SHA256_CHECKSUM}"
+                      CMAKE_ARGS ${AWSSDK_CMAKE_ARGS}
+                      BUILD_BYPRODUCTS ${AWS_CRT_CPP_STATIC_LIBRARY}
+                      DEPENDS aws_c_auth_ep
+                              aws_c_cal_ep
+                              aws_c_common_ep
+                              aws_c_compression_ep
+                              aws_c_event_stream_ep
+                              aws_c_http_ep
+                              aws_c_io_ep
+                              aws_c_mqtt_ep
+                              aws_c_s3_ep
+                              aws_c_sdkutils_ep
+                              aws_checksums_ep
+                              aws_lc_ep
+                              s2n_ep)
+  add_dependencies(AWS::aws-crt-cpp aws_crt_cpp_ep)
 
   externalproject_add(awssdk_ep
                       ${EP_COMMON_OPTIONS}
                       URL ${AWSSDK_SOURCE_URL}
                       URL_HASH "SHA256=${ARROW_AWSSDK_BUILD_SHA256_CHECKSUM}"
                       CMAKE_ARGS ${AWSSDK_CMAKE_ARGS}
-                      PATCH_COMMAND ${AWSSDK_PATCH_COMMAND}
                       BUILD_BYPRODUCTS ${AWS_CPP_SDK_COGNITO_IDENTITY_STATIC_LIBRARY}
                                        ${AWS_CPP_SDK_CORE_STATIC_LIBRARY}
                                        ${AWS_CPP_SDK_IDENTITY_MANAGEMENT_STATIC_LIBRARY}
                                        ${AWS_CPP_SDK_S3_STATIC_LIBRARY}
                                        ${AWS_CPP_SDK_STS_STATIC_LIBRARY}
-                      DEPENDS aws_c_event_stream_ep)
+                      DEPENDS aws_c_event_stream_ep aws_crt_cpp_ep)
```

Can we remove `aws_c_event_stream_ep` here because `aws_crt_cpp_ep` depends on `aws_c_event_stream_ep`?

```suggestion
                      DEPENDS aws_crt_cpp_ep)
```

---

done